Network requests implemented

Pavel-Laukhin opened the pull request:
Домашнее задание номер 2 в целом выполнено.
Единственное, что смущает - не удалось сделать так, чтобы скрол вью увеличивался и весь контент сдвигался, если клавиатура загораживает поля ввода на странице с параметрами запроса.

Pavel-Laukhin committed:
CocoaPods with Kingfisher dependency has been added.

Pavel-Laukhin committed:
GitHub logo added

Pavel-Laukhin committed:
Rounded SignIn button

Pavel-Laukhin committed:
1. HelloViewController added.

Pavel-Laukhin committed:
1. URLRequest handler added.
2. Avatar image has been made rounded.
3. Added possibility to choose the number of stars for adding into request.

Pavel-Laukhin committed:
Tried to negate obscuring text fields by the keyboard. Unfortunately, unsuccessful. :[

nskazakov commented on GitHubClient/HelloViewController.swift:
```diff
@@ -0,0 +1,254 @@
+        label.text = "Hello"
+        label.font = UIFont.systemFont(ofSize: 30, weight: .bold)
+        return label
+    }()
```
добавь отступ в одну строку между твоими созданными вьюхами

nskazakov commented on GitHubClient/HelloViewController.swift:
```diff
@@ -0,0 +1,254 @@
+            self.scrollView.scrollIndicatorInsets = insets
+            self.scrollView.scrollRectToVisible(self.starsNumberTextField.frame, animated: true)
+        }
+        keyboardWillChangeFrameObserver = NotificationCenter.default.addObserver(forName: UIResponder.keyboardWillChangeFrameNotification, object: nil, queue: .main, using: keyboardWillChangeFrameClosure)
```
попробуй добавить UIVIew.animate и перзагрузить layout
self.view.layoutIfNeeded()

Pavel-Laukhin committed:
Added layoutIfNeeded() method calling within registerForKeyboardWillChangeFrameNotification(). It still doesn't work, even if it's being wrapped into UIView.animate

Pavel-Laukhin committed:
The problem of obscuring content with the keyboard has been resolved. Thanks to Aleksey Bardin's magic! :]

Pavel-Laukhin committed:
Added space lines between properties in HelloViewController.

Pavel-Laukhin committed:
Changed scroll view behavior and keyboard observers in HelloViewController. Now the content of the screen scrolls automatically after the editing starts and you can scroll the content as you want.

Pavel-Laukhin committed:
1. Added ActivityIndicatorViewController.
2. Replaced keyboard observers in HelloViewController to Nikita's. They have be overhauled for self-scrolling.
3. Added ResultTableViewController and its ResultTableViewCell.
4. Rebuild QueryEngine for ability to serialize JSON and to show table with search results.
5. Added Repo structure.
6. Made some architecture changes.

Pavel-Laukhin committed:
Changed keyboard observers.
